### PR TITLE
QA-925 attempt to fix test failures not sent to big query

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/QA-925

cc @gdlex4015 

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSuite.scala [hunk 1]
```diff
@@ -23,6 +25,7 @@ trait GPAllocFixtureSpec extends fixture.FreeSpecLike with Retries {
 
     sys.props.get(gpallocProjectKey) match {
       case None => throw new RuntimeException("leonardo.billingProject system property is not set")
+      case Some(msg) if msg.startsWith("Failed To Claim Project") => throw new RuntimeException(msg)
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSuite.scala:28]: maybe `if msg.toLowerCase.startsWith("failed to claim project")`?

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSuite.scala [hunk 2]
```diff
@@ -41,8 +44,10 @@ trait GPAllocBeforeAndAfterAll extends BeforeAndAfterAll with BillingFixtures wi
 
   override def beforeAll(): Unit = {
     super.beforeAll()
-    val billingProject = claimProject()
-    sys.props.put(gpallocProjectKey, billingProject.value)
+    Either.catchNonFatal(claimProject()) match {
+      case Left(e) => sys.props.put(gpallocProjectKey, "Failed To Claim Project: "+e.getMessage)
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSuite.scala:48]: Oh I see that's our message - thought it came from an external library or something.

**Review by rtitle — APPROVED** [review]
The change looks fine but I think I saw some of these issues happen because cluster creation failed in `ClusterFixtureSpec.beforeAll`. Not sure how to make ScalaTest report those kinds of errors.

**Comment by Qi77Qi** [comment]
@rtitle added https://github.com/DataBiosphere/leonardo/pull/1176/commits/75304331ac83d1142914f3a9cac2d7d0ba6f8031

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/ClusterFixtureSpec.scala [hunk 3]
```diff
@@ -114,8 +119,12 @@ abstract class ClusterFixtureSpec extends fixture.FreeSpec with BeforeAndAfterAl
     if (!debug) {
       sys.props.get(gpallocProjectKey) match {
         case Some(billingProject) =>
-          createRonCluster(GoogleProject(billingProject))
-        case None => throw new RuntimeException("leonardo.billingProject system property is not set")
+          Either.catchNonFatal(createRonCluster(GoogleProject(billingProject))).handleError {
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/ClusterFixtureSpec.scala:122]: The exception might be `AssertionError` -- does that match Scala `NonFatal`?

**Comment by rtitle** [comment]
👍 sure worth a try, thanks

**Comment by Qi77Qi** [comment]
@rtitle I think it's probably okay. These exceptions are fatal.
```
     case _: VirtualMachineError | _: ThreadDeath | _: InterruptedException | _: LinkageError | _: ControlThrowable => false
```

**Pull request merged**